src/sync/installation.ts

```diff
@@ -313,25 +314,41 @@ async function doProcessInstallation(app, data: BackfillMessagePayload, sentry:
 		statsd.increment(metricTaskStatus.complete, [`type: ${nextTask.task}`]);
 
 	} catch (err) {
-		await handleBackfillError(err, logger, scheduleNextTask, data, task, repositoryId, subscription, nextTask);
+		await handleBackfillError(err, data, nextTask, subscription, logger, scheduleNextTask);
 	}
 }
 
-const handleBackfillError = async (err,
-	logger: LoggerWithTarget,
-	scheduleNextTask: (delayMs) => void, data: BackfillMessagePayload,
-	task: "pull" | "commit" | "branch",
-	repositoryId: string,
+/**
+ * Handles an error and takes action based on the error type and parameters
+ *
+ * @param err The error
+ * @param logger
+ * @param scheduleNextTask Function which schedules next task with the specified delay
+ * @param ignoreCurrentRepo Function which ignores currently processing repo and schedules the next task
+ * @param failCurrentRepoAndContinue Function which sets the status of the current repo sync to "failed" and schedules the next task
+ */
+export const handleBackfillError = async (err,
+	data: BackfillMessagePayload,
+	nextTask: Task,
 	subscription: SubscriptionClass,
-	nextTask: Task) : Promise<void> => {
+	logger: LoggerWithTarget,
+	scheduleNextTask: (delayMs: number) => void) : Promise<void> => {
+
+	const isRateLimitError = (err instanceof RateLimitingError || err instanceof OldRateLimitingError) || Number(err?.headers?.["x-ratelimit-remaining"]) == 0;
 
-	const rateLimit = (err instanceof RateLimitingError || err instanceof OldRateLimitingError) ? err.rateLimitReset : Number(err?.headers?.["x-ratelimit-reset"]);
-	const delay = Math.max(rateLimit * 1000 - Date.now(), 0);
+	if(isRateLimitError) {
+		const rateLimit = (err instanceof RateLimitingError || err instanceof OldRateLimitingError) ? err.rateLimitReset : Number(err?.headers?.["x-ratelimit-reset"]);
+		const delay = Math.max(rateLimit * 1000 - Date.now(), 0);
 
-	if (delay) {
-		// if not NaN or 0
-		logger.info({ delay }, `Delaying job for ${delay}ms`);
-		scheduleNextTask(delay);
+		if (delay) {
+			// if not NaN or 0
+			logger.info({delay}, `Delaying job for ${delay}ms`);
+			scheduleNextTask(delay);
+		} else {
+			//Retry immediately if rate limiting reset already
+			logger.info("Rate limit was reset already. Scheduling next task")
+			scheduleNextTask(0);
+		}
```

Comment:
/nit `scheduleNextTask(delay)` and `scheduleNextTask(0)` is the same thing so long as `rateLimit` defaults to 0, or you do  `scheduleNextTask(delay || 0)`

Reply:
Yes, true

Reply:
I tried to refactor, but I'd prefer to keep it like this, it seems more readable to me to make it explicit.

```
		if (delay) {
			// if not NaN or 0
			logger.info({delay}, `Delaying job for ${delay}ms`);
		} else {
			//Retry immediately if rate limiting reset already
			logger.info("Rate limit was reset already. Scheduling next task")
		}
		scheduleNextTask(delay);
		return;
```